File: 7_reverse_integer/main.cpp
#include <iostream>
#include <limits.h>

class Solution {
public:
    int reverse(int x) {
        long res = 0;
        while (x != 0) {
            res = res * 10 + x % 10;
            if (res > INT_MAX || res < INT_MIN) return 0;
            x /= 10;
        }
        return res;
    }
};

int main() {
    Solution s;
    int x;

    x = 123;
    std::cout << s.reverse(x) << std::endl;

    x = -123;
    std::cout << s.reverse(x) << std::endl;

    x = 120;
    std::cout << s.reverse(x) << std::endl;
    
    return 0;
}

File: 7_reverse_integer/go/main.go
package main

import (
	"fmt"
)

func reverse(x int) int {
	res := 0
	for x != 0 {
		res = res * 10 + x % 10
		if res > 2147483647 || res < -2147483648 {
			return 0
		}
		x /= 10
	}
	return res
}

func main() {
	var x int

	x = 123
	fmt.Println(reverse(x))

	x = -321
	fmt.Println(reverse(x))

	x = 120
	fmt.Println(reverse(x))
}
